draft-ietf-tls-esni.md

```diff
@@ -482,6 +469,39 @@ would divulge the true server name. The remaining
 contents of the ClientHelloOuter MAY be identical to those in
 ClientHelloInner but MAY also differ.
 
+## Recommended Padding Scheme {#padding}
+
+This section describes a deterministic padding mechanism based on the following
+observation: individual extensions can reveal sensitive information through their
+length. Thus, each extension in the inner ClientHello may require different amounts
+of padding. This padding may be fully determined by the client's configuration or
+may require server input.
+
+By way of example, a client's set of ALPN values is typically fixed. Clients SHOULD
+pad this extension by rounding up to the longest of those values. The target padding
+length of most ClientHello extensions can be computed in this way.
```

Comment:
This text seems a bit confusing because clients typically offer a fixed set of ALPNs for a given application.  Perhaps:

"Clients typically support a small number of application profiles. For instance, a browser might support HTTP with ALPN values ["http/1.1, "h2"] and WebRTC media with ALPNs ["webrtc", "c-webrtc"]. Clients SHOULD pad this extension by rounding up to the total size of the longest ALPN extension across all application profiles".

Reply:
This might be a little narrow a view.  In a browser, we have a bunch of things we might do.  But that is because the browser is platform.  But the anonymity set on a different platform (say you used the system TLS stack) is potentially wider (all apps) and cannot make the same claim about application profiles.

Reply:
The point is that the "platform," be it the stack used in the browser or in the OS, typically defines the set of values here and should pad accordingly. I'm fine with using the browser as an example here, as it's just a specific instance of that platform.